05-JWT-Basics/preferred/public/index.js

```diff
@@ -0,0 +1,53 @@
+const form = document.querySelector('.form');
+const submitButton = document.querySelector('.submit-button'); 
+const nameInput = document.querySelector('.name-input');
+const passwordInput = document.querySelector('.password-input');
+const loginMessage = document.querySelector('.login-message'); 
+const authButton = document.querySelector('.auth-button')
+const authMessage = document.querySelector('.auth-message');
+const successMessage = document.querySelector('.success');
+
+loginMessage.hidden = true; 
+successMessage.hidden = true; 
+
+form.addEventListener('submit', async (e) => {
+  e.preventDefault(); 
+  const name = nameInput.value;
+  const password = passwordInput.value; 
+  console.log('submit', name, password); 
+
+  if (!name || !password) {
+    loginMessage.hidden = false;
+  } 
```

Comment:
In this case, it looks like you'd want to just do a clean `return` out of the function or something similar. That's so none of the below function code executes (since the user provided no username and/or no password).